### PR TITLE
Moving docker environment to docker-compose

### PR DESCRIPTION
**Pull request by RPierini**
The objective of this PR is to move the environment instantiation process to a docker-compose approach, allowing the use of Secrets to configure the SSHparameters and keys at runtime, without the need to rebuild the Docker image with every change in keys or configs, nor leaving the keys and configs saved in the layers of the build process. This will allow for distribution of the image, which will be configured by each developer individually with their own keys.

### CODE REVIEW

File: docker/build_docker [hunk 1]
```diff
@@ -22,11 +22,11 @@ main() {
   echo "Base image "${base_dev_image}""
 
   # Copy configuration files and ssh keys to the docker context folder.
-  cp "${GIT_CONFIG}" "${context}"
-  cp "${SSH_FOLDER}"/"${SSH_PRIVATE_KEY_FILE}" "${context}"
-  cp "${SSH_FOLDER}"/"${SSH_PUBLIC_KEY_FILE}" "${context}"
-  cp "${SSH_FOLDER}"/"${SSH_CONFIG_FILE}" "${context}"
-  cp "${BASH_RC}" "${context}"
+  #cp "${GIT_CONFIG}" "${context}"
```
Comment (navarrothiago) [inline docker/build_docker:25]: Remove old code

File: docker/build_docker [hunk 2]
```diff
@@ -42,11 +42,11 @@ main() {
                --build-arg BASE_DEV_IMAGE="${base_dev_image}"
 
   # Remove configuration files and ssh keys from the docker context folder.
-  rm "${context}"/.gitconfig
-  rm "${context}"/"${SSH_PRIVATE_KEY_FILE}"
-  rm "${context}"/"${SSH_PUBLIC_KEY_FILE}"
-  rm "${context}"/"${SSH_CONFIG_FILE}"
-  rm "${context}"/.bashrc
+  #rm "${context}"/.gitconfig
```
Comment (navarrothiago) [inline docker/build_docker:45]: Removes

File: docker/run_docker [hunk 3]
```diff
@@ -8,12 +17,30 @@ main() {
   local -r dirname="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
   source "${dirname}"/../env.sh
 
-	docker run -it --rm \
-		--env-file "${dirname}"/../env.sh \
-		--volume "${dirname}"/../:"${WORKSPACE}" \
-		--privileged \
-		--workdir "${WORKSPACE}" "${IMAGE_TAG}":"${IMAGE_VERSION}" \
-		/bin/bash
+	# docker run -it --rm \
```
Comment (navarrothiago) [inline docker/run_docker:20]: removes

**Review by navarrothiago — COMMENTED** [review]
Thanks a lot for the contribution. Just few comments.

File: docker/build_docker (hunk 2 above)
Comment (RPierini) [reply docker/build_docker:45]: solved

File: docker/run_docker (hunk 3 above)
Comment (RPierini) [reply docker/run_docker:20]: solved

**Review by RPierini — COMMENTED** [review]
comments solved, old code removed

**Pull request merged**